### DISCUSSION TITLE
Are people building their typescript packages?

### DISCUSSION THREAD

**Discussion by reggi** [vercel/turborepo#257] [Help] [answered]
I have this question up on SO https://stackoverflow.com/questions/70342151/including-local-package-in-webpack-build-turborepo

I'm struggling with getting webpack to recognize the ui package I setup. 

I'm wondering if it's worth it to just setup a bundler for my ui library so it dist's a built js package so I don't have to worry about upstream compilation. 

Is this the traditional method? I really didn't want to have to have another watched folder, but I suppose webpack wont hotrelad this code anyway? 

What's the best way to build code using the a turborepo setup with typescript / webpack?

**Comment by anthonyshew** [comment, marked as answer]
We recommend the [Internal Package pattern](https://turbo.build/repo/docs/handbook/sharing-code/internal-packages) for packages that are meant to be used within your monorepo. If you're publishing there are more intricacies that we've written about [here](https://turbo.build/repo/docs/handbook/publishing-packages).
